### DISCUSSION TITLE
Does anyone update dependency in template projects?

### DISCUSSION THREAD

**Discussion by artemptushkin** [renovatebot/renovate#27870] [Suggest an Idea]
### Tell us more.

Let's say we have a project with files with placeholders, e.g [ejs](https://ejs.co/) files

build.gradle.ejs

```ejs
// some gradle stuff
sonarqube {
    properties {
        property("sonar.projectKey", "<%= projectName %>")
    }
}
```

**How can we automate the dependencies updates in them?**

Options (they require details, I keep the description short):

* Run the generation before the Renovate
  * cons:
    * we have to move all the renovate changes from the generated files to the `.ejs` files as they are in the repo
* Run Renovate on client-side, whenever we run the template engine after that we run Renovate
  * cons:
    * small: it's okay in the pipeline if Renovate runs remotely, it's a problem if you generate a project locally - you will run Renovate locally with a different set up than in the pipeline

-----

* [Related EJS discussion](https://github.com/mde/ejs/issues/753)

**Comment by artemptushkin** [comment]
okay, my case was that we have `.ejs` files and [Renovate simply didn't support them](https://github.com/renovatebot/renovate/blob/11b3c59215f49e033e45a243bce10a7d1c918711/docs/usage/java.md?plain=1#L34) if you use classic `build.gradle` etc. files it works even if they contain placeholders

**Reply by viceice** [reply]
you can use regex manager to work around
